src/Core/Python/PythonDatatypeConverter.cc

```diff
@@ -93,6 +94,25 @@ namespace
   }
 }
 
+boost::python::dict SCIRun::Core::Python::wrapDatatypesInMap(const std::vector<Datatypes::MatrixHandle>& matrices,
+  const std::vector<FieldHandle>& fields, const std::vector<Datatypes::StringHandle>& strings)
+{
+  boost::python::dict d;
+  int i = 0;
+  for (const auto& m : matrices)
+  {
+    if (auto dense = castMatrix::toDense(m))
+      d[i++] = convertMatrixToPython(dense);
+    else if (auto sparse = castMatrix::toSparse(m))
+      d[i++] = convertMatrixToPython(sparse);
+  }
+  for (const auto& f : fields)
+    d[i++] = convertFieldToPython(f);
+  for (const auto& s : strings)
```

Comment:
Good opportunity to invoke move semantics if you don't need copy of the strings:

`for (auto && s : strings)`

Reply:
That line by itself won't move anything; the function that uses the `s` value would need to move it, and it doesn't it just reads it and builds a new object, which means an r-value reference/move semantics won't gain anything.